### PR TITLE
Add collection property to STAC Items (#384)

### PR DESCRIPTION
**Pull request by m-mohr**
As discussed in #384, this is the PR to add a `collection` property to STAC Items at the root-level. 

Having #415 in mind, do we specifically need to add a collection filter field to the API or does it just work out of the box?

### CODE REVIEW

**Comment by matthewhanson** [comment]
I'm good with this change, but yes we do want to add a new search term for querying against collection. Want to add it this PR @m-mohr ?

**Comment by matthewhanson** [comment]
Maybe add the change to make `commons` extension core this PR as well @m-mohr ?

**Comment by m-mohr** [comment]
> yes we do want to add a new search term for querying against collection. Want to add it this PR @m-mohr ?

@matthewhanson Yes, should be included here. Directly belongs together, I think. I'm not so familiar with the search API parts yet. Could you help with it? I'm currently not sure where to put it.

> Maybe add the change to make `commons` extension core this PR as well @m-mohr ?

These are related changes, but can be done in separate PRs IMHO. I currently don't have the time to make the commons extension core, maybe will only do during the June sprint. So let's split them. It's also easier to review them separately, I guess.

File: item-spec/item-spec.md [hunk 1]
```diff
@@ -36,15 +36,16 @@ incorporated.
...
+| properties | Properties Object                                                          | **REQUIRED.** A dictionary of additional metadata for the item. |
+| links      | [Link Object]                                                              | **REQUIRED.** List of link objects to resources and related URLs. A link with the `rel` set to `self` is required. |
+| assets     | Map<string, Asset Object>                                                  | **REQUIRED.** Dictionary of asset objects that can be downloaded, each with a unique key. Some pre-defined keys are listed in the chapter 'Asset types'. |
+| collection | string                                                                     | The `id` of the STAC Collection this Item references to (see `collection` relation type below). This field is *required* if such a relation type is present. This field provides an easy way for a user to search for any Items that belong in a specified Collection. |
```
Comment (kbgg) [inline item-spec/item-spec.md]: Here it says that this is a required field if the collection relation is present yet in the readme it states that this field is strongly recommended. These seem to be contradicting.

Comment (m-mohr) [reply item-spec/item-spec.md]: Thanks for spotting this, I'll check what we prefer.

**Comment by matthewhanson** [comment]
I'm adding my review to this as I think it's fine to merge, but for 0.7.0 since this change goes in we will also need to update query to include a collections parameter.

@cholmes where we left it (IIRC) was that were would add new `id` and `collection` query terms and leave the `query` term for querying properties only so you wouldn't have to include "properties." for every term.

The `fields` term on the other hand was going to require "properties." for including any properties.

**Comment by cholmes** [comment]
Ah, that's right @matthewhanson  I think there was general head nods on doing it for query too, but I think it's cool to hold off on that till next sprint.

Do we have `fields` term updated for that change? Or do we need an issue / PR for it too?

Will merge this PR.

File: item-spec/item-spec.md (hunk 1 above)
Comment (m-mohr) [reply item-spec/item-spec.md]: If I get you correct, than you are referring to the collection rel type mentioned in the Item spec (strongly recommended) and to the collection field/property for items. It makes sense the way it is: It is strongly recommended to link to a collection. If a link to a collection is in the Item, it is required to add a collection field/property.

Comment (kbgg) [reply item-spec/item-spec.md]: Ah, you're right! Sorry about that.

**Pull request merged**